src/slice-assign.c

```diff
@@ -31,19 +32,29 @@ SEXP vctrs_assign(SEXP x, SEXP index, SEXP value, SEXP x_arg_, SEXP value_arg_)
   struct vctrs_arg x_arg = new_wrapper_arg(NULL, r_chr_get_c_string(x_arg_, 0));
   struct vctrs_arg value_arg = new_wrapper_arg(NULL, r_chr_get_c_string(value_arg_, 0));
 
-  return vec_assign(x, index, value, &x_arg, &value_arg);
+  const struct vec_assign_opts opts = {
+    .assign_names = false,
+    .x_arg = &x_arg,
+    .value_arg = &value_arg
+  };
+
+  return vec_assign_opts(x, index, value, &opts);
 }
 
 // [[ include("vctrs.h") ]]
-SEXP vec_assign(SEXP x, SEXP index, SEXP value,
-                struct vctrs_arg* x_arg,
-                struct vctrs_arg* value_arg) {
+SEXP vec_assign(SEXP x, SEXP index, SEXP value) {
```

Comment:
I don't think we use this internally yet, but if we did I'd imagine that we'd want to pass `x_arg` and `value_arg` through?

Reply:
The idea was to keep the main function as easy to call as possible and to have another one with an option struct for additional parameters.